src/config/semrush-token-manager.php

```diff
@@ -0,0 +1,101 @@
+<?php namespace Yoast\WP\SEO\Config;
+
+use League\OAuth2\Client\Token\AccessTokenInterface;
+use Yoast\WP\SEO\Exceptions\SEMrush\SEMrush_Empty_Token_Property_Exception;
+use Yoast\WP\SEO\Exceptions\SEMrush\SEMrush_Failed_Token_Storage_Exception;
+use Yoast\WP\SEO\Values\SEMrush\SEMrush_Token;
+
+/**
+ * Class SEMrush_Token_Manager
+ * @package Yoast\WP\SEO\Config
+ */
+class SEMrush_Token_Manager {
```

Comment:
Not sure about the name of this class. `Manager` feels a tad too generic.

One idea would be to make this a `Repository`, also placing it in the same directory and thus making the value object it creates a model.

Reply:
Moving this code to a Repository / Model setup, would mean the token would have to be stored in a different manner than an option or not?

Reply:
It would be stored exactly the same. It's more a matter of general concepts.

Having a `Repository` be somewhat more generic than simply fetches and saves things from a table and instead be fetches and saves things to the database feels like a cleaner domain than introducing a new `Manager` concept.